### PR TITLE
Resolved: 1192-migrating-organizations-who-have-already-registered-on-iati-publ…

### PR DESCRIPTION
**Pull request by Sanilblank**
…isher

* [x] Created command that will be used to migrate the existing organization

### CODE REVIEW

**Review by sriza — CHANGES_REQUESTED** [review]

File: app/Console/Commands/MigrateExistingOrganizationCommand.php [hunk 1]
```diff
@@ -0,0 +1,218 @@
+<?php
+
```
Comment (sriza) [inline app/Console/Commands/MigrateExistingOrganizationCommand.php:2]: add strict type

File: app/IATI/Traits/LogFunctionTrait.php [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+ * Class LogFunctionTrait
+ */
+trait LogFunctionTrait
+{
+    /**
+     * Log Activity Not Published Because Of Token Verification.
+     *
+     * @throws PublishException
+     */
+    private function logActivityNotPublishedBecauseOfTokenVerification($aidStreamOrganization, $iatiOrganization, $activityPublished)
```
Comment (A4family) [inline app/IATI/Traits/LogFunctionTrait.php:20]: Return type not defined

File: app/IATI/Traits/LogFunctionTrait.php [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+        throw new PublishException($iatiOrganization->id, $message);
+    }
+
+    /**
+     * Log Activity Not Published Because Of Publisher Verification.
+     *
+     * @throws PublishException
+     */
+    private function logActivityNotPublishedBecauseOfPublisherVerification($aidStreamOrganization, $iatiOrganization, $activityPublished)
```
Comment (A4family) [inline app/IATI/Traits/LogFunctionTrait.php:40]: Return type not defined

File: app/IATI/Traits/LogFunctionTrait.php [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+        throw new PublishException($iatiOrganization->id, $message);
+    }
+
+    /**
+     * Log Activity Not Published Because Organization Setting Not Valid.
+     *
+     * @throws PublishException
+     */
+    private function logActivityNotPublishedBecauseOrganizationSettingNotValid($aidStreamOrganization, $iatiOrganization, $activityPublished)
```
Comment (A4family) [inline app/IATI/Traits/LogFunctionTrait.php:60]: return type not defined

File: app/IATI/Traits/LogFunctionTrait.php [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+     *
+     * @param $aidStreamOrganization
+     * @param $iatiOrganization
+     * @param $organizationPublished
+     *
+     * @return mixed
+     *
+     * @throws PublishException
+     */
+    private function logOrganizationNotPublishedBecauseTokenVerification($aidStreamOrganization, $iatiOrganization, $organizationPublished): mixed
```
Comment (A4family) [inline app/IATI/Traits/LogFunctionTrait.php:110]: return type should be void.

File: app/IATI/Traits/LogFunctionTrait.php [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+     *
+     * @param $aidStreamOrganization
+     * @param $iatiOrganization
+     * @param $organizationPublished
+     *
+     * @return mixed
+     *
+     * @throws PublishException
+     */
+    private function logOrganizationNotPublishedBecausePublisherVerification($aidStreamOrganization, $iatiOrganization, $organizationPublished): mixed
```
Comment (A4family) [inline app/IATI/Traits/LogFunctionTrait.php:137]: return type should be void.

File: app/IATI/Traits/LogFunctionTrait.php [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+        throw new PublishException($iatiOrganization->id, $message);
+    }
+
+    /**
+     * Log Organization Not Published Because Organization Setting Is Not Valid.
+     *
+     * @throws PublishException
+     */
+    private function logOrganizationNotPublishedBecauseOrganizationSettingIsNotValid($aidStreamOrganization, $iatiOrganization, $organizationPublished, $setting)
```
Comment (A4family) [inline app/IATI/Traits/LogFunctionTrait.php:159]: return type not defined

File: app/IATI/Traits/MigrateActivityTrait.php [hunk 8]
```diff
@@ -1476,4 +1481,190 @@ public function getConditions(?array $conditionArray): ?array
...
+        }
+    }
+
+    /**
+     * Set default values where empty.
+     *
+     * @param $iatiElement
+     * @param $aidStreamOrganizationSetting
+     * @param bool $activityLevel
+     * @return void
```
Comment (A4family) [inline app/IATI/Traits/MigrateActivityTrait.php:1526]: add new line for different phpdocs type

File: app/IATI/Traits/MigrateDocumentFileTrait.php [hunk 9]
```diff
@@ -72,4 +72,134 @@ public function getDocumentFiles($aidstreamOrganizationId): Collection
...
+    }
+
+    /**
+     * Migrates Aid stream document data to iati document table.
+     *
+     * @param $aidstreamOrganizationId
+     * @param $iatiOrganization
+     * @param $migratedActivitiesLookupTable
+     *
+     * @return void
```
Comment (A4family) [inline app/IATI/Traits/MigrateDocumentFileTrait.php:110]: add a new line between @return and @throws

**Review by sriza — APPROVED** [review]

**Pull request merged**